src/app/component/auth/components/restore-password/restore-password.component.ts

```diff
@@ -82,17 +82,15 @@ export class RestorePasswordComponent implements OnInit, OnDestroy {
   }
 
   private onSignInFailure(errors: HttpErrorResponse): void {
-    try {
-      errors.error.forEach(error => {
-        if (error.name === 'email') {
-          this.emailErrorMessageBackEnd = error.message;
-        } else if (error.name === 'password') {
-          this.passwordErrorMessageBackEnd = error.message;
-        }
-      });
-    } catch (e) {
-      this.backEndError = errors.error.message;
-    }
+  if (!Array.isArray(errors.error)) {
+   this.backEndError = errors.error.message;
```

Comment:
Explain the logic of your checking.

Reply:
I'm checking if we get array of errors. If false - assign in this.backEndError
                                                            If true - another check